src/client/components/hud/Initiative.tsx

```diff
@@ -252,17 +259,82 @@ export function InitiativeHUD() {
 
   const focusTokenOnTurnStart = useRRSettings()[0].focusTokenOnTurnStart;
 
+  const updateConcentration = useCallback(
+    (
+      character: RRCharacter,
+      updater: React.SetStateAction<RRCharacter["currentlyConcentratingOn"]>
+    ) => {
+      dispatch((state) => {
+        const old =
+          state.characters.entities[character.id]?.currentlyConcentratingOn;
+
+        if (old === undefined) return [];
+        const newConcentration =
+          typeof updater === "function" ? updater(old) : updater;
+
+        return {
+          actions: [
+            characterUpdate({
+              id: character.id,
+              changes: { currentlyConcentratingOn: newConcentration },
+            }),
+          ],
+          optimisticKey: "concentration",
+          syncToServerThrottle: 0,
+        };
+      });
+    },
+    [dispatch]
+  );
+
+  const alertPlayer = useCallback(
+    async (characterName: string, concentrationObject: string) => {
+      await alert(
+        `${characterName}'s concentration on ${concentrationObject} has expired.`
+      );
+    },
+    [alert]
+  );
+
   useEffect(() => {
     if (
-      focusTokenOnTurnStart &&
       currentRow &&
       currentRow.id !== lastRowId.current &&
       currentRow.type === "character"
     ) {
-      setSelection(currentRow.characterIds);
+      if (focusTokenOnTurnStart) setSelection(currentRow.characterIds);
+      currentRow.characterIds.forEach((characterId) => {
+        const character = charactersRef.current.entities[characterId];
+        if (
+          character?.currentlyConcentratingOn &&
+          character.currentlyConcentratingOn.roundsLeft <= 1
+        ) {
+          if (
+            myCharacters.includes(character.id) ||
+            (character.localToMap && myself.isGM)
+          ) {
+            void alertPlayer(
+              character.name,
+              character.currentlyConcentratingOn.name
+            );
+          }
+          updateConcentration(character, null);
```

Comment:
I feel like this call should ideally also be moved to the server. But I'm not sure how we would trigger the alert then. But maybe at least add a TODO here?
I think the issue right now is that if a player is offline, then they'll never see the message and their concentrationRounds will be continually decremented. Once they come online again and it is their turn, they will get a message that their concentration expired, even though it has expired several rounds ago.